### PR TITLE
[6.0-staging] Update some Windows server helix queues for mono

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes some mono timeout and unexpected crash failures happening due to using obsolete images.

### CODE REVIEW

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 1]
```diff
@@ -130,13 +130,13 @@ jobs:
             - Windows.Amd64.Server2022.Open
             - ${{ if ne(parameters.jobParameters.runtimeFlavor, 'mono') }}:
               - (Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64
-              - (Windows.Server.Core.1909.Amd64.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-2004-helix-amd64
+              -  (Windows.Amd64.Server2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
```
Comment (carlossanlop) [inline eng/pipelines/libraries/helix-queues-setup.yml:133]: I think the prefix should match what we use in line 130. Correct me if I'm wrong.

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 2]
```diff
@@ -130,13 +130,13 @@ jobs:
...
           - ${{ if ne(parameters.jobParameters.isFullMatrix, true) }}:
             - Windows.81.Amd64.Open
             - Windows.Amd64.Server2022.Open
             - Windows.11.Amd64.Client.Open
             - ${{ if eq(parameters.jobParameters.testScope, 'outerloop') }}:
-              - (Windows.Server.Core.1909.Amd64.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-2004-helix-amd64
+              -  (Windows.Amd64.Server2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
```
Comment (hoyosjs) [inline eng/pipelines/libraries/helix-queues-setup.yml:139]:
```suggestion
              - (Windows.Amd64.Server2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
```

File: eng/pipelines/libraries/helix-queues-setup.yml (hunk 1 above)
Comment (hoyosjs) [reply eng/pipelines/libraries/helix-queues-setup.yml:133]:
```suggestion
              - (Windows.Server.Core.ltsc2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
```

The prefix is the friendly name - so the information about what container environment you are in is infinitely more useful.

File: eng/pipelines/libraries/helix-queues-setup.yml (hunk 2 above)
Comment (hoyosjs) [inline eng/pipelines/libraries/helix-queues-setup.yml:139]:
```suggestion
              -  (Windows.Server.Core.ltsc2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
```

**Review by hoyosjs — CHANGES_REQUESTED** [review]

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 3]
```diff
@@ -130,13 +130,13 @@ jobs:
...
               - (Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64
-              - (Windows.Server.Core.1909.Amd64.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-2004-helix-amd64
+              - (Windows.Server.Core.ltsc2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
           - ${{ if ne(parameters.jobParameters.isFullMatrix, true) }}:
             - Windows.81.Amd64.Open
             - Windows.Amd64.Server2022.Open
             - Windows.11.Amd64.Client.Open
             - ${{ if eq(parameters.jobParameters.testScope, 'outerloop') }}:
-              - (Windows.Server.Core.1909.Amd64.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-2004-helix-amd64
+              -  (Windows.Server.Core.ltsc2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
```
Comment (carlossanlop) [inline eng/pipelines/libraries/helix-queues-setup.yml:139]:
```suggestion
              - (Windows.Server.Core.ltsc2022.Open)windows.amd64.server2022.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-amd64
```

**Review by akoeplinger — APPROVED** [review]

**Review by hoyosjs — APPROVED** [review]

**Pull request merged**